docs/html/topics/vcs-support.md

```diff
@@ -0,0 +1,163 @@
+# VCS Support
+
+pip supports installing from various version control systems (VCS).
+This support requires a working executable to be available (for the version
+control system being used). It is used through URL prefixes:
+
+- Git -- `git+`
+- Mercurial -- `hg+`
+- Subversion -- `svn+`
+- Bazaar -- `bzr+`
+
+## Supported VCS
+
+### Git
+
+The supported schemes are `git+file`, `git+https`, `git+ssh`, `git+http`,
+`git+git` and `git`. Here are some of the supported forms:
+
+```none
+git+ssh://git.example.com/MyProject#egg=MyProject
+git+file:///home/user/projects/MyProject#egg=MyProject
+git+https://git.example.com/MyProject#egg=MyProject
+```
+
+```{warning}
+The use of `git`, `git+git`, and `git+http` schemes is discouraged.
+The former two use [the Git Protocol], which lacks authentication, and HTTP is
+insecure due to lack of TLS based encryption.
+```
+
+[the Git Protocol]: https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols
+
+It is also possible to specify a "git ref" such as branch name, a commit hash or
+a tag name:
+
+```none
+git+https://git.example.com/MyProject.git@master#egg=MyProject
+git+https://git.example.com/MyProject.git@v1.0#egg=MyProject
+git+https://git.example.com/MyProject.git@da39a3ee5e6b4b0d3255bfef95601890afd80709#egg=MyProject
+git+https://git.example.com/MyProject.git@refs/pull/123/head#egg=MyProject
+```
+
+When passing a commit hash, specifying a full hash is preferable to a partial
+hash because a full hash allows pip to operate more efficiently (e.g. by
+making fewer network calls).
+
+### Mercurial
+
+The supported schemes are `hg+file`, `hg+http`, `hg+https`, `hg+ssh`
+and `hg+static-http`. Here are some of the supported forms:
+
+```
+hg+http://hg.myproject.org/MyProject#egg=MyProject
+hg+https://hg.myproject.org/MyProject#egg=MyProject
+hg+ssh://hg.myproject.org/MyProject#egg=MyProject
+hg+file:///home/user/projects/MyProject#egg=MyProject
+```
+
+It is also possible to specify a revision number, a revision hash, a tag name
+or a local branch name:
+
+```none
+hg+http://hg.example.com/MyProject@da39a3ee5e6b#egg=MyProject
+hg+http://hg.example.com/MyProject@2019#egg=MyProject
+hg+http://hg.example.com/MyProject@v1.0#egg=MyProject
+hg+http://hg.example.com/MyProject@special_feature#egg=MyProject
+```
+
+### Subversion
+
+The supported schemes are `svn`, `svn+svn`, `svn+http`, `svn+https` and
+`svn+ssh`. Here are some of the supported forms:
+
+```none
+svn+https://svn.example.com/MyProject#egg=MyProject
+svn+ssh://svn.example.com/MyProject#egg=MyProject
+svn+ssh://user@svn.example.com/MyProject#egg=MyProject
+```
+
+You can also give specific revisions to an SVN URL, like so:
+
+```none
+-e svn+http://svn.example.com/svn/MyProject/trunk@2019#egg=MyProject
+-e svn+http://svn.example.com/svn/MyProject/trunk@{20080101}#egg=MyProject
+```
+
+Note that you need to use [Editable VCS installs](#editable-vcs-installs) for
+using specific revisions.
```

Comment:
These paragraphs seem out of place. Why are they only present for SVN, not other VCS sections?

Reply:
Yup, only subversion. I dunno why, but that's how the code works. 🤷🏻‍♂️

I'll tweak this sentence to clarify that.